drivers/espi/espi_mchp_xec.c

```diff
@@ -0,0 +1,822 @@
+ *  40h   | SMVW03 | res          | res          | DNX_ACK   | SUS_ACK#    |
+ *  41h   | MSVW03 | SLP_A#       | res          | SUS_PDNACK| SUS_WARN#   |
+ *  42h   | MSVW04 | res          | res          | SLP_WLAN# | SLP_LAN#    |
+ *  43h   | MSVW05 | generic      | generic      | generic   | generic     |
+ *  44h   | MSVW06 | generic      | generic      | generic   | generic     |
+ *  45h   | SMVW04 | generic      | generic      | generic   | generic     |
+ *  46h   | SMVW05 | generic      | generic      | generic   | generic     |
+ *  47h   | MSVW07 | res          | res          | res       | HOST_C10    |
+ *  4Ah   | MSVW08 | res          | res          | DNX_WARN  | res         |
+ */
+
+static const struct xec_signal vw_tbl[] = {
+	/* MSVW00 */
+	[ESPI_VWIRE_SIGNAL_SLP_S3]        = { MSVW00, SRC_ID0, M2S },
+	[ESPI_VWIRE_SIGNAL_SLP_S4]        = { MSVW00, SRC_ID1, M2S },
+	[ESPI_VWIRE_SIGNAL_SLP_S5]        = { MSVW00, SRC_ID2, M2S },
+	/* MSVW01 */
+	[ESPI_VWIRE_SIGNAL_SUS_STAT]      = { MSVW01, SRC_ID0, M2S },
+	[ESPI_VWIRE_SIGNAL_PLTRST]        = { MSVW01, SRC_ID1, M2S },
+	[ESPI_VWIRE_SIGNAL_OOB_RST_WARN]  = { MSVW01, SRC_ID2, M2S },
+	/* SMVW00 */
+	[ESPI_VWIRE_SIGNAL_OOB_RST_ACK]   = { SMVW00, SRC_ID0, S2M },
+	[ESPI_VWIRE_SIGNAL_WAKE]          = { SMVW00, SRC_ID2, S2M },
+	[ESPI_VWIRE_SIGNAL_PME]           = { SMVW00, SRC_ID3, S2M },
+	/* SMVW01 */
+	[ESPI_VWIRE_SIGNAL_SLV_BOOT_DONE] = { SMVW01, SRC_ID0, S2M },
+	[ESPI_VWIRE_SIGNAL_ERR_FATAL]     = { SMVW01, SRC_ID1, S2M },
+	[ESPI_VWIRE_SIGNAL_ERR_NON_FATAL] = { SMVW01, SRC_ID2, S2M },
+	[ESPI_VWIRE_SIGNAL_SLV_BOOT_STS]  = { SMVW01, SRC_ID3, S2M },
+	/* SMVW02 */
+	[ESPI_VWIRE_SIGNAL_SCI]           = { SMVW02, SRC_ID0, S2M },
+	[ESPI_VWIRE_SIGNAL_SMI]           = { SMVW02, SRC_ID1, S2M },
+	[ESPI_VWIRE_SIGNAL_RST_CPU_INIT]  = { SMVW02, SRC_ID2, S2M },
+	[ESPI_VWIRE_SIGNAL_HOST_RST_ACK]  = { SMVW02, SRC_ID3, S2M },
+	/* MSVW02 */
+	[ESPI_VWIRE_SIGNAL_HOST_RST_WARN] = { MSVW02, SRC_ID0, M2S },
+	/* SMVW03 */
+	[ESPI_VWIRE_SIGNAL_SUS_ACK]       = { SMVW03, SRC_ID0, S2M },
+	[ESPI_VWIRE_SIGNAL_DNX_ACK]       = { SMVW03, SRC_ID1, S2M },
+	/* MSVW03 */
+	[ESPI_VWIRE_SIGNAL_SUS_WARN]      = { MSVW03, SRC_ID0, M2S },
+	[ESPI_VWIRE_SIGNAL_SUS_PWRDN_ACK] = { MSVW03, SRC_ID1, M2S },
+	[ESPI_VWIRE_SIGNAL_SLP_A]         = { MSVW03, SRC_ID3, M2S },
+	/* MSVW04 */
+	[ESPI_VWIRE_SIGNAL_SLP_LAN]       = { MSVW04, SRC_ID0, M2S },
+	[ESPI_VWIRE_SIGNAL_SLP_WLAN]      = { MSVW04, SRC_ID1, M2S },
+	/* MSVW07 */
+	[ESPI_VWIRE_SIGNAL_HOST_C10]      = { MSVW07, SRC_ID0, M2S },
+	/* MSVW08 */
+	[ESPI_VWIRE_SIGNAL_DNX_WARN]      = { MSVW08, SRC_ID1, M2S },
+};
+
+static int espi_xec_configure(struct device *dev, struct espi_cfg *cfg)
+{
+	u8_t iomode = 0;
+	u8_t cap1 = ESPI_CAP_REGS->GLB_CAP1;
+	u8_t cur_iomode = (cap1 & MCHP_ESPI_GBL_CAP1_IO_MODE_MASK) >>
+		MCHP_ESPI_GBL_CAP1_IO_MODE_POS;
+
+	/* Set frequency */
+	cap1 &= ~MCHP_ESPI_GBL_CAP1_MAX_FREQ_MASK;
+	switch (cfg->max_freq) {
```

Comment:
have an empty line before the switch. it has to be coalesced if only the previous assignement is being used in the if ()/for ()/while ()/ switch ()

Reply:
same here

Reply:
Addressed